docs/run.js

```diff
@@ -97,29 +97,61 @@ function drawTableAndSetUpHandlers(processedData, map, scoreAsset) {
  * @param {number|string} index checkbox index, basis for id
  * @param {String} displayName checkbox display name
  * @param {div} parentDiv div to attach checkbox to
+ * @param {Object} colorFunction color data from the layer
  * @return {HTMLInputElement} the checkbox
  */
 function createNewCheckbox(index, displayName, parentDiv) {
   const newRow = document.createElement('div');
   newRow.className = 'checkbox-row';
+
   const newBox = document.createElement('input');
   newBox.type = 'checkbox';
   newBox.id = getCheckBoxId(index);
-  newBox.className = 'checkbox';
+  newBox.className = 'checkbox layer-checkbox';
   newBox.checked = true;
   newRow.appendChild(newBox);
-  const newMark = document.createElement('span');
-  newMark.className = 'checkmark';
-  newRow.appendChild(newMark);
+
   const label = document.createElement('label');
   label.className = 'checkbox-label';
   label.htmlFor = newBox.id;
   label.innerHTML = displayName;
   newRow.appendChild(label);
+
   parentDiv.appendChild(newRow);
   return newBox;
 }
 
+/**
+ * Gets the linear gradient of the colors for the legend.
+ *
+ * @param {Object} colorFunction color data from the layer
+ * @return {string} the linear gradient
+ */
+function getLinearGradient(colorFunction) {
+  if (!colorFunction) {
+    return '';
+  }
+  const colors = [];
+  const currentStyle = colorFunction['current-style'];
+  switch (currentStyle) {
+    case 0:
+      colors.push('white', colorFunction['color']);
```

Comment:
I guess this is so that we can tell the difference between single-color and continuous? I'm not sure it's adding value, though, especially since it's then unclear on the difference between discrete and continuous. If you do want to distinguish, I'd suggest doing discrete vertically and continuous horizontally.